test/integration_test.py

```diff
@@ -386,13 +386,26 @@ def test_catalog_listing(self):
 
     def test_snapshot_listing(self):
         """
-        Test with a small page size to be sure paging works
+        Test with small page sizes to make sure paging works.
         """
-        page_size = 5
-        with mock.patch.object(TDRClient, 'page_size', page_size):
-            paged_snapshots = self._public_tdr_client.snapshot_names_by_id()
-        snapshots = self._public_tdr_client.snapshot_names_by_id()
-        self.assertEqual(snapshots, paged_snapshots)
+        # Without a filter, the test takes so long that there's a real risk of
+        # failure due to new snapshots being added mid-test.
+        #
+        snapshot_filters_by_deployment = {
+            'anvildev': 'anvil_',  # ~5 snapshots
+            'dev': 'hca_dev_5',  # ~10 snapshots
+            'anvilprod': 'anvil_page_',  # ~13 snapshots
+            'prod': '_dcp37'  # ~13 snapshots
+        }
+        filter = snapshot_filters_by_deployment[config.main_deployment_stage]
+        for page_size in 1, 2:
+            with self.subTest(page_size=page_size):
+                with mock.patch.object(TDRClient, 'page_size', page_size):
+                    paged_snapshots = self._public_tdr_client.snapshot_names_by_id(filter=filter)
+                    self.assertGreaterEqual(self._public_tdr_client.page_size, 1)
+                snapshots = self._public_tdr_client.snapshot_names_by_id(filter=filter)
+                self.assertLess(len(snapshots), 20)
```

Comment:
A limit of 20 snapshots was chosen as a candidate constraint. This is a generous 54% increase, compared with the total number of snapshots (13) expected to be observed in stable deployments (highest snapshot usage). This limit also implicitly places an upper bound on the number of round-trips to TDR. With the combined page sizes tested, the total number of round-trips required is constrained at roughly 32, in the worst case. Any snapshot count exceeding this limit in stable deployments, or elsewhere, should warrant further refining of the filters used, to keep this test specific and efficient.

Reply:
Please raise in PL.

Reply:
The number of snapshots used in this test is limited to 20, bounding the amount of round-trip requests to TDR to 32, in total for this test. I picked 20 as the limit of snapshots because it seemed a sufficient margin over 13, the current maximum count of snapshots returned for this filter.